src/main/java/io/cdap/plugin/http/source/common/RawStringPerLine.java

```diff
@@ -61,14 +65,26 @@ public boolean hasNext() {
             isLineRead = true;
             return lastLine != null;
         } catch (IOException e) { // we need to catch this, since hasNext() does not have "throws" in parent
-            throw new RuntimeException("Failed to read line from http page buffer", e);
+            ErrorUtils.ActionErrorPair pair = ErrorUtils.getActionErrorByStatusCode(httpResponse.getStatusCode());
```

Comment:
similar comment here, please check first `httpResponseCode` is not `200`. Then only do this stuff.